webserver/Models/Controllers/RealEstatesController.cs

```diff
@@ -0,0 +1,154 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using webserver.Data;
+using webserver.Models;
+using webserver.Models.DTOs;
+
+namespace webserver.Controllers;
+
+[ApiController]
+[Route("api/v1/realestates")]
+public class RealEstatesController : ControllerBase {
+    
+    private readonly WebserverContext _context;
+
+    public RealEstatesController(WebserverContext context) {
+        _context = context;
+    }
+
+    [HttpGet("unique/{id}")]
+    public IActionResult ReadRealEstate(int id){
+
+        var realEstate = _context.RealEstates.Find(id);
+        if(realEstate==null){
+            return NotFound();
+        }
+
+        var response = new {
+            data = new {
+                type = "RealEstate",
+                attribute = new RealEstateDTO (id, realEstate.Name, realEstate.Address, realEstate.Price, realEstate.CompanyId)
+            }
+        };
+
+        return Ok(response);
+    }
+
+    [HttpGet]
+    public async Task<IActionResult> ReadRealEstates(int? minPrice, int? maxPrice, int? offset, int? limit, string sort = "Id") {
```

Comment:
You are implementing this manually.
I think you can use this library: https://www.jsonapi.net/

Reply:
Will check it out, thanks!